module/proto/peggy/v1/confirm.proto

```diff
@@ -0,0 +1,66 @@
+syntax = "proto3";
+package peggy.v1;
+
+option go_package = "github.com/althea-net/peggy/module/x/peggy/types";
+
+// ConfirmType is the cosmos type of a confirmation of a given object from the eth signers
+enum ConfirmType {
+  CONFIRM_TYPE_UNSPECIFIED = 0;
+  CONFIRM_TYPE_VALSET      = 1;
+  CONFIRM_TYPE_BATCH       = 2;
+  CONFIRM_TYPE_LOGIC       = 3;
+}
+
+// ConfirmLogicCall
+// When validators observe a MsgRequestBatch they form a batch by ordering
+// transactions currently in the txqueue in order of highest to lowest fee,
+// cutting off when the batch either reaches a hardcoded maximum size (to be
+// decided, probably around 100) or when transactions stop being profitable
+// (TODO determine this without nondeterminism) This message includes the batch
+// as well as an Ethereum signature over this batch by the validator
+// -------------
+message ConfirmLogicCall {
+  string invalidation_id    = 1;
+  uint64 invalidation_nonce = 2;
+  string eth_signer         = 3;
+  string orchestrator       = 4;
```

Comment:
nit, can we rename these to `eth_address` and `orchestrator_address`?

Reply:
agreed